web/crux/src/builders/notification.template.builder.ts

```diff
@@ -0,0 +1,49 @@
+import { Injectable } from '@nestjs/common'
+import { readFileSync } from 'fs'
+import { join } from 'path'
+import { cwd } from 'process'
+import Handlebars from 'handlebars'
+import { InvalidArgumentException } from 'src/exception/errors'
+
+type TemplateEntry = {
+  text?: string
+  file?: string
+}
+
+type TemplateFile = { [key: string]: TemplateEntry }
+
+export type TemplateType = 'email' | 'chat'
+
+export type ProcessedTemplate = { [key: string]: string }
+
+@Injectable()
+export default class NotificationTemplateBuilder {
+  private processTemplateEntry<T>(entry: TemplateEntry, data: T): string {
+    let templateText = entry.text
+    if (entry.file) {
+      templateText = readFileSync(join(cwd(), 'assets', 'email', entry.file), 'utf8')
+    }
+
+    if (!templateText) {
+      throw new InvalidArgumentException({
+        property: 'template',
+        value: 'undefined',
+        message: `Invalid template`,
+      })
+    }
+
+    return Handlebars.compile(templateText)(data)
```

Comment:
"I am not very familiar with Handlebars, but what if it throws an exception? Isn't it necessary to have:

```
    try {
      return Handlebars.compile(templateText)(data);
    } catch (error) {
      throw new Error(`Could not compile template: ${error.message}`);
    }
    ```